File: selection_sort/selection-sort.js
//
// Selection Sort algorithm from Algorithms book chapter 2
//
//

function selectionSort(a) {
  let n = a.length;

  for (let i=0; i < n; i++) {
    let min = i;
    for (let j=i+1; j < n; j++) {
      if (a[j] < a[min]) {
        let temp = a[min]
        a[min] = a[j]
        a[j] = temp
      }
    }
  }
  console.log(a);
  return a
}

selectionSort([10, 9, 8, 7, 6, 5, 4, 3, 2, 1]);

File: selection_sort/selection_sort.rb
#
# Selection sort from chapter 2 in Algorithms book
#
require 'pry'

def selection_sort(a)
  n = a.length
  i = 0

  while i < n do
    min = i
    j = i + 1
    while j < n
      if a[j] < a[min]
        temp = a[min]
        a[min] = a[j]
        a[j] = temp
      end
      j += 1
    end
    i += 1
  end
  
  puts a
end

selection_sort([4,5,3,7,1,2,8,10])
